qabelbox/src/main/java/de/qabel/android/config/AndroidPersistence.java

```diff
@@ -0,0 +1,187 @@
+import de.qabel.core.exceptions.QblInvalidEncryptionKeyException;
+
+public class AndroidPersistence extends Persistence<QblSQLiteParams> {
+
+    private static final String STR_BLOB = "BLOB";
+    private static final String STR_ID = "ID";
+    private static final String STR_ID_QUERY = "ID = ?";
+    private QblSQLiteOpenHelper dbHelper;
+    private SQLiteDatabase database;
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(AndroidPersistence.class.getName());
+
+    public AndroidPersistence(QblSQLiteParams params) throws QblInvalidEncryptionKeyException {
+        connect(params);
+    }
+
+    @Override
+    public boolean connect(QblSQLiteParams params) {
+        dbHelper = new QblSQLiteOpenHelper(params.getContext(), params.getName(),
+                params.getFactory(), params.getVersion());
+        database = dbHelper.getWritableDatabase();
+        return true;
+    }
+
+    @Override
+    public boolean persistEntity(Persistable object) {
+        if (object == null) {
+            throw new IllegalArgumentException("Arguments cannot be null!");
+        }
+
+        String sql = "CREATE TABLE IF NOT EXISTS " +
+                getTableNameForClass(object.getClass()) +
+                "(ID TEXT PRIMARY KEY NOT NULL," +
+                "BLOB BLOB NOT NULL)";
+
+        try {
+            database.execSQL(sql);
+        } catch (SQLException e) {
+            LOGGER.error("Cannot create table!", e);
+        }
+
+        ContentValues contentValues = new ContentValues();
+        contentValues.put(STR_ID, object.getPersistenceID());
+        contentValues.put(STR_BLOB, serialize(object.getPersistenceID(), object));
+
+        return database.insert(getTableNameForClass(object.getClass()), null, contentValues) != -1L;
+    }
+
+    @Override
+    public boolean updateEntity(Persistable object) {
+        if (object == null) {
+            throw new IllegalArgumentException("Arguments cannot be null!");
+        }
+
+        if (getEntity(object.getPersistenceID(), object.getClass()) == null) {
+            LOGGER.info("Entity not stored!");
+            return false;
+        }
+
+        ContentValues contentValues = new ContentValues();
+        contentValues.put(STR_BLOB, serialize(object.getPersistenceID(), object));
+
+        String[] whereArgs = { object.getPersistenceID() };
+        return database.update(getTableNameForClass(object.getClass()),
+                contentValues, STR_ID_QUERY, whereArgs) != -1;
+    }
+
+    @Override
+    public boolean updateOrPersistEntity(Persistable object) {
+        if (getEntity(object.getPersistenceID(), object.getClass()) == null) {
+            return persistEntity(object);
+        }
+        else {
+            return updateEntity(object);
+        }
+    }
+
+    @Override
+    public boolean removeEntity(String id, Class cls) {
+        if (id == null || cls == null) {
+            throw new IllegalArgumentException("Arguments cannot be null!");
```

Comment:
defensive programming. You may discuss this with @audax but imho this is simply bad